
syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.raft";
option java_outer_classname = "RaftProto";
option objc_class_prefix = "HLW";

package raft;

service dummyRaftService{
    rpc send_message(MessageContent) returns (MessageReply){}
    rpc vote_request(VoteRequest) returns (VoteReply){}
}

message VoteRequest{
    int32 node_id = 1;
    string node_status = 2;
    int32 term_no = 3;
}

message VoteReply{
    int32 term_no = 1;
    bool vote_granted = 2;
}

message MessageContent {
    int32 node_id = 1;
    string node_status = 2;
    int32 term_no = 3;
}

message MessageReply{
    int32 term_no = 1;
}